news/8372.removal

```diff
@@ -0,0 +1,4 @@
+Deprecate -b/--build/--build-dir/--build-directory. Its current behaviour
```

Comment:
I wouldn't say that the reason is the new resolver, but that it's behavior is confusing to begin with and doing it via temp dir + --no-clean is much better for all the known usecases.

This was surfaced while trying to implement for the new resolver, and is something we'd prefer to clean up instead of replicating bug-for-bug.

Reply:
@pradyunsg I reworded the changelog entry.